db/schema_tables.cc

```diff
@@ -1772,7 +1791,27 @@ lw_shared_ptr<keyspace_metadata> create_keyspace_from_schema_partition(const sch
     auto strategy_name = strategy_options["class"];
     strategy_options.erase("class");
     bool durable_writes = row.get_nonnull<bool>("durable_writes");
-    return make_lw_shared<keyspace_metadata>(keyspace_name, strategy_name, strategy_options, durable_writes);
+
+    data_dictionary::storage_options storage_opts;
+    // Scylla-specific info is fetched below
+    if (features.contains<schema_feature::SCYLLA_KEYSPACES>()) {
+        auto keyspace_key = dht::decorate_key(*scylla_keyspaces(), partition_key::from_singular(*scylla_keyspaces(), keyspace_name));
+        auto&& rs = co_await db::system_keyspace::query(proxy, NAME, SCYLLA_KEYSPACES, keyspace_key);
```

Comment:
This makes the function non-pure. Can we move it to the caller?

Reply:
Technically we could, and that was my initial approach, but I dropped because there are several callers and **not** fetching this info would be a bug, which would be extremely trivial to miss if I suddenly start requiring the caller to do it. It would be extremely easy to miss some of the changes during backports. I gave it some thought and ultimately decided that if I absolutely require something to be called before this function, it's better to just add it to the function body, so that nobody can forget to do it. If you still prefer me to try and split it, let me know.

Reply:
update: I'll try to slightly refactor this function's signature to try and get rid of this co_await. I'll let you know if I succeeded or not, and if not I'll just come back here and discuss why.

Reply:
done!